packages/mobile/routes/onboarding/views/storage-protection-setup/views/SetupPinProtectionView.svelte

```diff
@@ -0,0 +1,151 @@
+    import { validatePinFormat } from '@lib/utils'
+    import { HTMLButtonType } from 'shared/components/Button.svelte'
+    import { onMount } from 'svelte'
+
+    export let busy = false
+
+    let setPinInput = ''
+    let setPinInputError = ''
+    let confirmPinInput = ''
+    let confirmPinInputError = ''
+    let arePinInputsMatching = false
+    let arePinInputsValid = false
+    let confirmPinInputElement: PinInput
+    let submitButtonElement: Button
+
+    $: setPinInput, (setPinInputError = '')
+    $: confirmPinInput, (confirmPinInputError = '')
+    $: arePinInputsMatching = setPinInput === confirmPinInput
+    $: arePinInputsValid = validatePinFormat(setPinInput) && validatePinFormat(confirmPinInput)
+    $: if (arePinInputsValid && !arePinInputsMatching) {
+        confirmPinInputError = localize('error.pincode.match')
+    } else {
+        confirmPinInputError = ''
+    }
+
+    function onBackClick(): void {
+        if ($isOnboardingLedgerProfile) {
+            /**
+             * CAUTION: We must make sure to stop polling if the user
+             * goes back as we've started it when this view is mounted.
+             */
+            stopPollingLedgerNanoStatus()
+        }
+        $storageProtectionSetupRouter.previous()
+    }
+
+    async function onSetPinClick(): Promise<void> {
+        resetPinInputErrors()
+        if (arePinInputsValid && arePinInputsMatching) {
+            await handleSetPin()
+        }
+    }
+
+    function resetPinInputErrors(): void {
+        setPinInputError = ''
+        confirmPinInputError = ''
+    }
+
+    async function handleSetPin(): Promise<void> {
+        busy = true
+
+        await initialisePincodeManager(setPinInput)
+
+        const canInitialiseFirstShimmerClaimingAccount =
+            $onboardingProfile?.type === ProfileType.Software &&
+            $onboardingProfile?.setupType === ProfileSetupType.Claimed
+        const shouldInitialiseFirstShimmerClaimingAccount = $onboardingProfile?.shimmerClaimingAccounts?.length < 1
+        if (canInitialiseFirstShimmerClaimingAccount && shouldInitialiseFirstShimmerClaimingAccount) {
+            await initialiseFirstShimmerClaimingAccount()
+        }
+
+        busy = false
+
+        $storageProtectionSetupRouter.next()
+    }
+
+    onMount(() => {
+        /**
+         * NOTE: We begin Ledger Nano status polling
+         * here because it's the closest common view between
+         * all Ledger flows that comes before the status
+         * check page, improving the UX as the status will
+         * already have been set by that point rather than setting
+         * it on mount.
+         */
+        if ($isOnboardingLedgerProfile) {
+            pollLedgerNanoStatus()
+        }
+    })
+</script>
+
+<OnboardingLayout {onBackClick} {busy}>
+    <div slot="title">
+        <Text type="h2">{localize('views.onboarding.storageProtectionSetup.setupPinProtection.title')}</Text>
+    </div>
+    ´
```

Comment:
there seems to be an extra character here

Reply:
ups 😅